lib/bolt/config.rb

```diff
@@ -39,6 +41,7 @@ def initialize(transport, uri = nil)
 
     DEFAULTS = {
       concurrency: 100,
+      'compile-concurrency': Concurrent.processor_count,
```

Comment:
Should we add a check to `validate` to ensure the value is less than or equal to this? We can error or warn + use processor count if it's too high

Reply:
:woman_shrugging:  I don't expect users will try to run with compile-concurrency of 0, but didn't realize there was value to setting it above the cpu count. Prolly fine as is

Reply:
Added validation anyway.